imagebuildah/stage_executor.go

```diff
@@ -2246,6 +2246,7 @@ func (s *StageExecutor) pullCache(ctx context.Context, cacheKey string) (referen
 // intermediateImageExists returns true if an intermediate image of currNode exists in the image store from a previous build.
```

Comment:
This godoc needs to be updated: the method returns the ID of a matching image, and if there are multiple matches, it returns the one with the most recent "created" date.